src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -294,10 +294,19 @@ function useChatUtilsHook(): UseChatUtilsHook {
     return pushMessage;
   };
 
+  const setSelectedChannel = (channel: ChannelList) => {
+    setChat((prevChat) => ({
+      ...prevChat,
+      selectedChannel: channel,
+      isLoadingFetchingChannelDetail: false
+    }));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `setSelectedChannel` function is a good addition as it allows for updating the selected channel in the chat state. However, it's important to note that this function directly mutates the state. If there are any side effects or asynchronous operations associated with changing the selected channel, they should be handled appropriately to avoid potential issues.



<!-- This is an auto-generated comment by OSS CodeRabbit -->